src/cloudai/schema/test_template/chakra_replay/slurm_command_gen_strategy.py

```diff
@@ -24,11 +24,7 @@ class ChakraReplaySlurmCommandGenStrategy(SlurmCommandGenStrategy):
     """Command generation strategy for ChakraReplay on Slurm systems."""
 
     def gen_exec_command(self, tr: TestRun) -> str:
```

Comment:
This change effectively makes `gen_exec_command` == `_write_sbatch_script`. I questionable:
1. `gen_exec_command` is a wrapper to run a "pipeline". Subclasses can override details of this pipeline, but it still consists of several part.
2. `_write_sbatch_script` is complicated enough already to include even more logic. And it is harder to test this function one, need to mock or rely on other functions' logic.

I do like that `_write_sbatch_script` now accepts `TestRun` though, it seems logic. And maybe we could apply it all other "building blocks" of `gen_exec_command` keeping it as a wrapper. What do you think?

Reply:
> This change effectively makes gen_exec_command == _write_sbatch_script.

It is true for most of the test templates, but not all of them. NeMo launcher does not use _write_sbatch_script for gen_exec_command.

Are you proposing to merge _write_sbatch_script into gen_exec_command? This may simplify _write_sbatch_script, but it would make it harder for us to differentiate between NeMo launcher's gen_exec_command and others' gen_exec_command, which seems essential to me. Moreover, I believe the current complexity of _write_sbatch_script is acceptable."

Reply:
No, I'm not suggesting merging these two functions. But I'm not sold on the idea of putting more logic into _write_sbatch_script either. So honestly not sure if this PR in current state makes sense.

Reply:
I'm trying to understand why
```py
def gen_exec_cmd(...):
    return self._write_sbatch_script(...)

def _write_sbatch_script(...):
    env_vars = self._override_env_vars(...)
    cmd_args = self._override_cmd_args(...)
    slurm_args = self._parse_slurm_args(...)
    srun_command = self.generate_srun_command(...)

    # file writing logic
```
is better compared to
```py
def gen_exec_cmd(...):
    env_vars = self._override_env_vars(...)
    cmd_args = self._override_cmd_args(...)
    slurm_args = self._parse_slurm_args(...)
    srun_command = self.generate_srun_command(...)
    return self._write_sbatch_script(...)

def _write_sbatch_script(...):
    # file writing logic
```

Second option looks better from logic isolation point of view.

Reply:
> Second option looks better from logic isolation point of view.

I agree that this makes sense.

Let me clarify my intentions. First, if we take the second approach, the function calls will be duplicated across test templates (e.g., _override_env_vars, _override_cmd_args, etc.). Second, I am considering this from the perspective of the plugin implementation. Following the current implementation would require calling plugin-related functions in gen_exec_cmd, exacerbating code duplication. My goal is to allow users to override functions like _parse_slurm_args and generate_srun_command as needed in the templates. I also plan to remove or simplify _override_env_vars and _override_cmd_args.